### PR TITLE
Fix profile picture in social logins

### PR DESCRIPTION
**Pull request by jakubfijalkowski**
In short - Apple does not include `picture` claim at all (at least according to docs) and we can get if we have a silhouette for FB.

### CODE REVIEW

File: src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookClient.cs [hunk 1]
```diff
@@ -120,6 +121,20 @@ private FacebookUser ConvertResponse(JsonElement result)
                     return null;
                 }
             }
+
+            static bool GetIsSilhouette(JsonElement element)
+            {
+                if (element.TryGetProperty("picture", out var pic) &&
+                    pic.TryGetProperty("data", out var data) &&
```
Comment (Saancreed) [inline src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookClient.cs:128]: Unused? Though I'm not sure if there is any reason to check for `picture` and `data` unless it's possible to receive a response that doesn't have them but has `"is_silhouette": true`. 
```suggestion
                if (element.TryGetProperty("picture", out _) &&
                    pic.TryGetProperty("data", out _) &&
```

**Review by Saancreed — APPROVED** [review]
Looks like you found a piece of documentation I couldn't have :see_no_evil:

One minor suggestion, LGTM otherwise.

File: src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookUser.cs [hunk 2]
```diff
@@ -7,14 +7,22 @@ public sealed class FacebookUser
         public string? FirstName { get; }
         public string? LastName { get; }
         public string Photo { get; }
+        public bool IsSilhouette { get; }
 
-        public FacebookUser(string id, string? email, string? firstName, string? lastName, string photo)
+        public FacebookUser(
```
Comment (Saancreed) [inline src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookUser.cs:12]: By the way, why isn't this a `record` like other user types?

Comment (lukaszgarstecki) [reply src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookUser.cs:12]: I would guess that this is leftover after FacebookModule 🕯️

**Review by lukaszgarstecki — APPROVED** [review]

File: src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookClient.cs (hunk 1 above)
Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookClient.cs:128]: Read closely :P

That's the response:
```
{
  "id": "4922594101147747",
  "name": "Jakub Fijałkowski",
  "picture": {
    "data": {
      "height": 50,
      "is_silhouette": true,
      "url": "https://scontent-waw1-1.xx.fbcdn.net/v/t1.30497-1/cp0/c15.0.50.50a/p50x50/84628273_176159830277856_972693363922829312_n.jpg?_nc_cat=1&ccb=3&_nc_sid=12b3be&_nc_ohc=-8z4hHC_XtoAX_HAa9v&_nc_ht=scontent-waw1-1.xx&tp=27&oh=c69e9a83d261021e2a01a272bb50f618&oe=6062A8B8",
      "width": 50
    }
  }
}
```

Comment (Saancreed) [reply src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookClient.cs:128]: Ah, damn, I'm blind :stuck_out_tongue:

File: src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookUser.cs (hunk 2 above)
Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.ExternalIdentityProviders/Facebook/FacebookUser.cs:12]: I will make it consistent in a separate PR (it needs some more work).

**Pull request merged**